src/native/managed/compile-native.proj

```diff
@@ -0,0 +1,52 @@
+        <SupportsNativeAotComponents Condition="'$(SupportsNativeAotComponents)' == '' and ('$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'armel' or '$(TargetArchitecture)' == 'x86' or '$(TargetArchitecture)' == 'riscv64')">false</SupportsNativeAotComponents>
+        <SupportsNativeAotComponents Condition="'$(SupportsNativeAotComponents)' == '' and ('$(TargetsWindows)' == 'true' or '$(TargetsOSX)' == 'true' or ('$(TargetsLinux)' == 'true' and '$(TargetsAndroid)' != 'true' and '$(TargetsLinuxMusl)' != 'true'))">true</SupportsNativeAotComponents>
+        <SupportsNativeAotComponents Condition="'$(SupportsNativeAotComponents)' == ''">false</SupportsNativeAotComponents>
+    </PropertyGroup>
+
+    <PropertyGroup>
+        <SysRoot Condition="'$(CrossBuild)' == 'true' and '$(HostOS)' != 'windows'">$(ROOTFS_DIR)</SysRoot>
+        <LinkerFlavor Condition="'$(CrossBuild)' == 'true' and '$(TargetsLinux)' == 'true'">lld</LinkerFlavor>
+        <CustomLinkerArgToolchainArg Condition="'$(CrossBuild)' == 'true' and '$(_hostArchitecture)' == '$(_targetArchitecture)' and '$(_hostOS)' != 'windows'">--gcc-toolchain=$(ROOTFS_DIR)/usr</CustomLinkerArgToolchainArg>
+        <ExtraProps Condition="'$(SysRoot)' != ''">$(ExtraProps);SysRoot=$(SysRoot)</ExtraProps>
+        <ExtraProps Condition="'$(LinkerFlavor)' != ''">$(ExtraProps);LinkerFlavor=$(LinkerFlavor)</ExtraProps>
+        <ExtraProps Condition="'$(CustomLinkerARgToolchainArg)' != ''">$(ExtraProps);CustomLinkerArgToolchainArg=$(CustomLinkerArgToolchainArg)</ExtraProps>
+    </PropertyGroup>
+
+    <ItemGroup Condition="$(SupportsNativeAotComponents)">
+        <ProjectReference Include="@(NativeLibsProjectsToBuild)"
+                          ReferenceOutputAssembly="false"
+                          AdditionalProperties="%(AdditionalProperties);RuntimeConfiguration=$(RuntimeConfiguration);LibrariesConfiguration=$(LibrariesConfiguration);RuntimeIdentifier=$(OutputRID);NativeLib=$(NativeLibKind)$(ExtraProps)"/>
+    </ItemGroup>
+
+    <Target Name="BuildNativeLibs" BeforeTargets="Build" Condition="$(SupportsNativeAotComponents)">
+        <Message Importance="High" Text="Restoring native libs @(NativeLibsProjectsToBuild)" />
+        <Message Importance="High" Text="Restoring native libs RuntimeConfiguration=$(RuntimeConfiguration) LibrariesConfiguration=$(LibrariesConfiguration)" />
+        <!-- workaround https://github.com/dotnet/msbuild/issues/2811 - Restore has to be a separate exectution with a different evaluation -->
+        <MSBuild Projects="@(NativeLibsProjectsToBuild)"
+                 Properties="Configuration=$(NativeLibsPublishConfiguration);RuntimeConfiguration=$(RuntimeConfiguration);LibrariesConfiguration=$(LibrariesConfiguration);RuntimeIdentifier=$(OutputRID);NativeLib=$(NativeLibKind)$(ExtraProps);_RandomPropertyToPleaseMSBuild=Restoring"
```

Comment:
We use the `_MSBuildRestoreSessionId` property as the special one for restore in other places, can we use that here as well?